bin/runtime/src/lib.rs

```diff
@@ -518,9 +518,32 @@ impl_runtime_apis! {
         }
     }
 
-    impl primitives::AlephApi<Block> for Runtime {
-        fn authorities() -> Vec<AlephId> {
-            Aleph::authorities()
+    impl primitives::AlephSessionApi<Block, AlephId, BlockNumber> for Runtime {
+        fn current_session() -> AuthoritySession<AlephId, BlockNumber> {
+            AuthoritySession {
+                session_id: Session::session_index(),
+                authorities: Aleph::authorities(),
+                stop_h: Session::estimate_next_new_session(System::block_number())
+                    .0
+                    .unwrap(),
+            }
+        }
+
+        fn next_session() -> AuthoritySession<AlephId, BlockNumber> {
+            AuthoritySession {
+                session_id: Session::session_index() + 1,
+                authorities: Session::queued_keys()
+                    .iter()
+                    .map(|(_, key)| key.get(AlephId::ID).unwrap_or_default())
```

Comment:
Such `unwrap_or_default` could hide some mistake and be hard to find. Usually, such calls are implemented in pallets so that an error as missing key here should be logged. Generally, `uwrap`s are good for debug or in places where you are 100% sure that it is really not `None`.

Reply:
I moved implementation to the palled and added error handling for this case. Other unwraps are ok and I added comments for them.